### ISSUE TITLE
Queries results in NoSuchMethodErrors with hibernate-reactive-core 2.0.0

### ISSUE THREAD

**Issue by gabfssilva** [line/kotlin-jdsl#272]
When executing simple queries, such as:

```kotlin
query.singleQueryOrNull<MyEntity> {
    select(entity(MyEntity::class))
    from(entity(MyEntity::class))
    where(col(MyEntity::id).equal(1))
}
```

A `NoSuchMethodError` is thrown, with the following details:

```
java.lang.NoSuchMethodError: 'org.hibernate.reactive.mutiny.Mutiny$Query org.hibernate.reactive.mutiny.Mutiny$StatelessSession.createQuery(jakarta.persistence.criteria.CriteriaQuery)'
```

It looks that `StatelessSession.createQuery` now returns a `SelectionQuery<R>`, while `HibernateMutinyReactiveQuery` expects a `Query<R>`. This seems to be causing the NoSuchMethodError.

More information:

- I'm using the `hibernate-reactive-kotlin-jdsl-jakarta:2.2.1` library.
- The non-jakarta module (`javax.persistence`) was working as expected.
- This library uses the `hibernate-reactive-core-jakarta` dependency, not the `hibernate-reactive-core`, so, this might be the source of the incompatibility.

**Comment by shouwn** [comment]
@cj848 Could you check this?

**Comment by superlabs-dev** [comment]
2.0 is currently unsupported. Because of the future roadmap of jdsl, I was wondering if I should support it. Think you need support? @shouwn

**Comment by shouwn** [comment]
I think I need support because it is a bug. I'm not sure because I don't fully understand the structure of the module on the reactive side, but it doesn't seem like a big change is needed.

I think it's better to fix it.

**Comment by gabfssilva** [comment]
This one looks a bit trickier than it appears. I thought it was just a library update + type adjusting, but the mutiny's query hierarchy has changed considerably: https://github.com/hibernate/hibernate-reactive/pull/1590

I'll take a closer look, but it seems like it's not possible to create an update query using the criteria builder anymore.

**Comment by cj848** [comment]
@shouwn @gabfssilva 
hibernate-reactive 2.0 seems to have minimum compatibility with hibernate 6.2 version. This is consistent with not being able to maintain javax.* packages anymore. It's not a simple matter. Either keep only the jakarta version of hibernate-reactive-core (actually, this is the most convenient, but not good from the point of view of maintaining compatibility), or support the maximum hibernate-reactive supported version of hibernate-reactive-core version to 1.x. And the hibernate-reactive-jakarta version must support 2.0
Please comment on which one is better.

-- korean
hibernate-reactive 2.0은 hibernate 6.2 버전이 최소 compatibility 로 보입니다. 이는 더이상 javax.* 패키지를 유지할 수 없다는것과 일치합니다. 간단한 문제는 아닙니다. hibernate-reactive-core 를 jakarta 버전만 유지하든지(사실 이게 제일 마음이 편하지만 호환성 유지 관점에서는 좋지 않습니다) 아니면 hibernate-reactive-core 버전의 최대 hibernate-reactive 지원 버전을 1.x 로 지원하든지 해야 합니다. 그리고 hibernate-reactive-jakarta 버전에서 2.0을 지원하는 형태가 되어야 합니다
어떤게 더 좋을지 의견 부탁드립니다.

**Comment by gabfssilva** [comment]
One thing that could potentially be done is to just make `hibernate-reactive-jakarta` use `hibernate-reactive-core:2.0.0`, while keeping the `hibernate-reactive` module with the previous version, so it would involve breaking changes only in the `hibernate-reactive-jakarta` module.

Based on the current facts, it seems these changes are unavoidable if one decides to upgrade to Hibernate 6.

**Comment by shouwn** [comment]
It's not as simple as I thought. I will abide by the decision of cj848. When it can be solved with minimal breakpoints, please change it. Since Kotlin JDSL 3.0 will create a huge breaking point, including an user interface, it seems good to be supported in 2.0 for users to use during migration from 2.0 to 3.0.

But if you think this breaking point is too big a range, it can be postponed to be resolved in Kotlin JDSL 3.0.
